addons/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxCommManager.java

```diff
@@ -0,0 +1,686 @@
+        if ((code >= 0x80) || ((code < 0x10) && (rawData[rawData.length - 3] == 0x43))) {
+            ackData = new byte[] { 0x0D, ackType, 0x43, 0x00, 0x0A };
+        } else {
+            ackData = new byte[] { 0x0D, ackType, 0x00, 0x0A };
+        }
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("sendAck(): sending message {}", DatatypeConverter.printHexBinary(ackData));
+        }
+        boolean done = sendMessage(ackData);
+        if (!done) {
+            logger.debug("sendAck(): failed");
+        }
+        return done;
+    }
+
+    /**
+     * Send a message to the Powermax alarm panel to change arm mode
+     *
+     * @param armMode the arm mode
+     * @param pinCode the PIN code. A string of 4 characters is expected
+     *
+     * @return true if the message was sent or false if not
+     */
+    public boolean requestArmMode(PowermaxArmMode armMode, String pinCode) {
+        logger.debug("requestArmMode(): armMode = {}", armMode.getShortName());
+
+        boolean done = false;
+        if (!armMode.isAllowedCommand()) {
+            logger.info("Powermax alarm binding: requested arm mode {} rejected", armMode.getShortName());
+        } else if ((pinCode == null) || (pinCode.length() != 4)) {
+            logger.info("Powermax alarm binding: requested arm mode {} rejected due to invalid PIN code",
+                    armMode.getShortName());
+        } else {
+            try {
+                byte[] dynPart = new byte[3];
+                dynPart[0] = armMode.getCommandCode();
+                dynPart[1] = (byte) Integer.parseInt(pinCode.substring(0, 2), 16);
+                dynPart[2] = (byte) Integer.parseInt(pinCode.substring(2, 4), 16);
+
+                done = sendMessage(new PowermaxBaseMessage(PowermaxSendType.ARM, dynPart), false, 0);
+            } catch (NumberFormatException e) {
+                logger.info("Powermax alarm binding: requested arm mode {} rejected due to invalid PIN code",
+                        armMode.getShortName());
+            }
+        }
+        return done;
+    }
+
+    /**
+     * Send a message to the Powermax alarm panel to change PGM or X10 zone state
+     *
+     * @param action the requested action. Allowed values are: OFF, ON, DIM, BRIGHT
+     * @param device the X10 device number. null is expected for PGM
+     *
+     * @return true if the message was sent or false if not
+     */
+    public boolean sendPGMX10(Command action, Byte device) {
+        logger.debug("sendPGMX10(): action = {}, device = {}", action, device);
+
+        boolean done = false;
+
+        Map<String, Byte> codes = new HashMap<>();
+        codes.put("OFF", (byte) 0x00);
+        codes.put("ON", (byte) 0x01);
+        codes.put("DIM", (byte) 0x0A);
+        codes.put("BRIGHT", (byte) 0x0B);
+
+        Byte code = codes.get(action.toString());
+        if (code == null) {
+            logger.info("Powermax alarm binding: invalid PGM/X10 command: {}", action);
```

Comment:
Internal classes should do no info logging - either there is a irrecoverable issue, then do error or warn with a stack trace - or else report any issues to the calling layers through exceptions and only do debug logging here